thewalrus/tests/test_samples.py

```diff
@@ -500,3 +501,26 @@ def test_out_of_bounds_generate_torontonian_sample():
     max_photons = 1
     samples = [generate_torontonian_sample(sigma, max_photons=max_photons) for i in range(n_samples)]
     assert -1 in samples
+
+
+def test_hafnian_sample_graph_rank_one():
```

Comment:
Would it be nice to include e.g. [flaky](https://github.com/box/flaky) in this test, or to mock out the random parts with `monkeypatch`? I'm thinking so that random fails don't occur in the future?